### PR TITLE
Small things all around

### PR DESCRIPTION
**Pull request by Malabarba**
- `cider-docview-file` will tell the user that it can't find the file, instead of throwing a cryptic error.
- `cider-jump-to` supports jumping to vars by name, besides position. This goes with a change in cider-nrepl which will provide us with the file even for variables whose origin we don't fully know (still, this commit is safe to merge on its own).
- Small fix and a defgroup on cider-debug.

### CODE REVIEW

File: cider-debug.el [hunk 1]
```diff
@@ -29,24 +29,31 @@
 (require 'cider-interaction)
 (require 'dash)
 
+
```
Comment (Malabarba) [inline cider-debug.el]: This is not an empty line, it's a ^L.

File: cider-doc.el [hunk 2]
```diff
@@ -175,10 +175,15 @@
   "Open the source for the current symbol, if available."
   (interactive)
   (if cider-docview-file
-      (let ((buffer (and cider-docview-file
-                         (not (cider--tooling-file-p cider-docview-file))
-                         (cider-find-file cider-docview-file))))
-        (cider-jump-to buffer (cons cider-docview-line nil) nil))
+      (-if-let (buffer (and (not (cider--tooling-file-p cider-docview-file))
+                            (cider-find-file cider-docview-file)))
+          (cider-jump-to buffer (if (> cider-docview-line 1)
```
Comment (Malabarba) [inline cider-doc.el]: I wasn't super happy with this part, but the repl reports a :line of 1 on defs evaluated with "eval", so the only way to check if we had line information was to check if it's `> 1`. Any better ideas?

**Comment by Malabarba** [comment]
This builds on #1176. So merge that first.

File: cider-doc.el (hunk 2 above)
Comment (bbatsov) [inline cider-doc.el]: Check whether the filename is a tooling file?

**Comment by bbatsov** [comment]
> cider-jump-to supports jumping to vars by name, besides position. This goes with a change in cider-nrepl which will provide us with the file even for variables whose origin we don't fully know (still, this commit is safe to merge on its own).

You should check that this works from stacktraces as well. Probably no extra changes are needed there, but still... better safe than sorry.

File: cider-doc.el (hunk 2 above)
Comment (Malabarba) [inline cider-doc.el]: Thanks to the upcoming changes to the `info` op, we won't get tooling files as filenames anymore (well, maybe sometimes, but much less often) so we can't use that to determine whether the line is correct or not.
However, you just made me realise, the `info` op itself could remove the line information whenever it manually replaces a tooling filename with a real filename. Do you forsee issues with that? (I'll push the actual `info` change soon)

**Comment by Malabarba** [comment]
I'll check stacktraces, I'll probably have to add a bit of code.

**Comment by Malabarba** [comment]
Stacktraces now work as well. And I've added a commit.
Also https://github.com/clojure-emacs/cider-nrepl/pull/223

**Comment by bbatsov** [comment]
> Stacktraces now work as well.

Nice. I guess we can mark #1175 as fixed and update the changelog.

**Comment by Malabarba** [comment]
Actually, I only tested that following buttons on Stack traces works, I forgot to test the highlighting. So don't close it just yet.

**Comment by Malabarba** [comment]
I just tested, and no, this is not enough to make highlighting work. I'm adding a comment to that discussion, but it's mosly unrelated to this commit, I think.

**Comment by Malabarba** [comment]
Reworded.

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**